zerver/views/__init__.py

```diff
@@ -91,9 +93,10 @@ def accounts_register(request):
         existing_user_profile = None
 
     validators.validate_email(email)
-
+    # If OPEN_REALM_CREATION is enabled all user sign ups should go through the
+    # sepcial URL with domain name so that REALM can be identified if multiple realms exist
     unique_open_realm = get_unique_open_realm()
-    if unique_open_realm:
+    if unique_open_realm is not None and not settings.OPEN_REALM_CREATION:
```

Comment:
OK.  I think it'd be better to have a more explicit signal for whether realm creation is happening as part of this request, and use that instead of the `realm is None` part of the conditional.  I think it'd make the code a lot clearer if we did it that way.